src/api/nabto_device_logging.c

```diff
@@ -20,8 +20,9 @@ static void logging_log(uint32_t severity, uint32_t module,
 void nabto_device_logging_init()
 {
     np_log.log = &logging_log;
-    logCallback = &default_log_callback;
-    userData = NULL;
+    if (!logCallback) {
```

Comment:
Why is it neccessary to set the logCallback here? this seems odd when the logCallback is global and the device init should probably not touch it.

Reply:
The logCallback must be set as the logger does not check before calling it. However, I can agree that it might as well be initialized to the default instead of NULL, and then init should not touch it.